### ISSUE TITLE
URGENT: tutorial does not work

### ISSUE THREAD

**Issue by wasade** [biocore/qiime#1847]
The tutorial data (this [URL](https://github.com/biocore/qiime/blob/master/examples/ipynb/illumina_overview_tutorial.ipynb#L49)) does not exist. It does not appear that this tutorial file is on the FTP. I'm marking this as urgent as this tutorial is going to be used at the Workshop on Genomics.

**Comment by ElDeveloper** [comment]
Just noticed that this URL was missing from the transition in #1823 yet it was still replaced to point to microbio.me i. e. the data was not re-downloaded but it was still updated.

**Comment by wasade** [comment]
This issue has been resolved, the data has been posted on the FTP. Thank you @ElDeveloper @gregcaporaso @jairideout and @antgonza for helping resolve this matter so quickly.

**Comment by wasade** [comment]
@gregcaporaso @jairideout, I'm reopening this issue as the QIIME 1.8 tutorial tarball does not reflect the QIIME 1.9 tutorial. Specifically, we're running into the following issue:

```bash

/usr/local/lib/python2.7/dist-packages/numpy/core/fromnumeric.py:2499: VisibleDeprecationWarning: `rank` is deprecated; use the `ndim` attribute or function instead. To find the rank of a matrix see `numpy.linalg.matrix_rank`.
  VisibleDeprecationWarning)
Error in validate_mapping_file.py: option -m: file does not exist: 'map.tsv'

If you need help with QIIME, see:
http://help.qiime.org
```

The mapping file appears to be in a google spreadsheet, but it needs to be contained in the tarball. We're going to recreate the file and move forward and see if there are any other issues.

**Comment by wasade** [comment]
Encountering this as well:

```bash
/usr/local/lib/python2.7/dist-packages/numpy/core/fromnumeric.py:2499: VisibleDeprecationWarning: `rank` is deprecated; use the `ndim` attribute or function instead. To find the rank of a matrix see `numpy.linalg.matrix_rank`.
  VisibleDeprecationWarning)
Error in validate_mapping_file.py: option -m: file does not exist: 'map-bad.tsv'

If you need help with QIIME, see:
http://help.qiime.org
```

**Comment by wasade** [comment]
`forward_reads.fastq.gz` and `barcodes.fastq.gz` do not exist either. Those can be recreated (AFAIK) from catting the existing raw data together

**Comment by wasade** [comment]
Getting a lot of these warnings:

```bash
/usr/local/lib/python2.7/dist-packages/numpy/core/fromnumeric.py:2499: VisibleDeprecationWarning: `rank` is deprecated; use the `ndim` attribute or function instead. To find the rank of a matrix see `numpy.linalg.matrix_rank`.
  VisibleDeprecationWarning)
```

The environment I'm using has numpy 1.9.1

**Comment by wasade** [comment]
`pick_open_reference_otus.py` requires HDF5:

```python
Traceback (most recent call last):
  File "/usr/local/bin/pick_open_reference_otus.py", line 450, in <module>
    main()
  File "/usr/local/bin/pick_open_reference_otus.py", line 429, in main
    minimum_failure_threshold=minimum_failure_threshold)
  File "/usr/local/lib/python2.7/dist-packages/qiime/workflow/pick_open_reference_otus.py", line 1076, in pick_subsampled_open_reference_otus
    table = Table.from_hdf5(biom_file)
  File "/usr/local/lib/python2.7/dist-packages/biom/table.py", line 3092, in from_hdf5
    raise RuntimeError("h5py is not in the environment, HDF5 support "
RuntimeError: h5py is not in the environment, HDF5 support is not available
```

I'm opening a separate issue for that

**Comment by wasade** [comment]
The deprecation warnings are due to using numpy 1.9.1. Currently, qiime requires numpy >= 1.9. I'm going to attempt to use numpy 1.8.2 which does not have these warnings as the volume of output stemming from these warnings will greatly confuse participants

**Comment by wasade** [comment]
With h5py 2.3.1 and 2.4.0, I now get the following:

```python
Traceback (most recent call last):
  File "/usr/local/bin/pick_open_reference_otus.py", line 450, in <module>
    main()
  File "/usr/local/bin/pick_open_reference_otus.py", line 429, in main
    minimum_failure_threshold=minimum_failure_threshold)
  File "/usr/local/lib/python2.7/dist-packages/qiime/workflow/pick_open_reference_otus.py", line 1076, in pick_subsampled_open_reference_otus
    table = Table.from_hdf5(biom_file)
  File "/usr/local/lib/python2.7/dist-packages/biom/table.py", line 3098, in from_hdf5
    id_ = h5grp.attrs['id']
AttributeError: 'file' object has no attribute 'attrs'
```

**Comment by wasade** [comment]
I'm at a blocking point with the tutorial and cannot continue. I'm not seeing any indication in the rc2 changes that the above issues have been encountered and addressed. Due to timing, and in order to provide the team here adequate time to familiarize themselves with the tutorial, I'm going to downgrade our environment to qiime 1.8.

The following test failures came from the tests for `pick_open_reference_otus.py`, which all boil down to testing methods that are not part of the default qiime install which is essentially what the environment is that I'm using.

```bash
======================================================================
ERROR: test_pick_subsampled_open_reference_otus_rdp_tax_assign (__main__.PickSubsampledReferenceOtusThroughOtuTableTests)
pick_subsampled_open_reference_otus fns when assigning tax with rdp
----------------------------------------------------------------------
Traceback (most recent call last):
  File "test_pick_open_reference_otus.py", line 364, in test_pick_subsampled_open_reference_otus_rdp_tax_assign
    minimum_failure_threshold=0)
  File "/usr/local/lib/python2.7/dist-packages/qiime/workflow/pick_open_reference_otus.py", line 1031, in pick_subsampled_open_reference_otus
    status_update_callback=status_update_callback)
  File "/usr/local/lib/python2.7/dist-packages/qiime/workflow/pick_open_reference_otus.py", line 233, in assign_tax
    close_logger_on_success=close_logger_on_success)
  File "/usr/local/lib/python2.7/dist-packages/qiime/workflow/util.py", line 122, in call_commands_serially
    raise WorkflowError(msg)
WorkflowError:

*** ERROR RAISED DURING STEP: Assign taxonomy
Command run was:
 assign_taxonomy.py -o /tmp/qiime_wf_outh0m4kp/rdp_assigned_taxonomy -i /tmp/qiime_wf_outh0m4kp/rep_set.fna --reference_seqs_fp /tmp/refseqsAduuUM --id_to_taxonomy_fp /tmp/refseqs_taxTOL1G4 --assignment_method rdp
Command returned exit status: 2
Stdout:

Stderr
/usr/local/lib/python2.7/dist-packages/numpy/core/fromnumeric.py:2499: VisibleDeprecationWarning: `rank` is deprecated; use the `ndim` attribute or function instead. To find the rank of a matrix see `numpy.linalg.matrix_rank`.
  VisibleDeprecationWarning)
Error in assign_taxonomy.py: RDP classifier is not installed or not accessible to QIIME. See install instructions here: http://qiime.org/install/install.html#rdp-install

If you need help with QIIME, see:
http://help.qiime.org



======================================================================
ERROR: test_pick_subsampled_open_reference_otus_sortmerna_sumaclust (__main__.PickSubsampledReferenceOtusThroughOtuTableTests)
pick_subsampled_open_reference_otus functions as expected with
----------------------------------------------------------------------
Traceback (most recent call last):
  File "test_pick_open_reference_otus.py", line 627, in test_pick_subsampled_open_reference_otus_sortmerna_sumaclust
    minimum_failure_threshold=100000)
  File "/usr/local/lib/python2.7/dist-packages/qiime/workflow/pick_open_reference_otus.py", line 715, in pick_subsampled_open_reference_otus
    close_logger_on_success=False)
  File "/usr/local/lib/python2.7/dist-packages/qiime/workflow/util.py", line 122, in call_commands_serially
    raise WorkflowError(msg)
WorkflowError:

*** ERROR RAISED DURING STEP: Pick Reference OTUs
Command run was:
 pick_otus.py -i /tmp/seqsH7cCLe -o /tmp/qiime_wf_outCJCuyJ/step1_otus -r /tmp/refseqsBgPA9S -m sortmerna  --suppress_new_clusters
Command returned exit status: 1
Stdout:

Stderr
/usr/local/lib/python2.7/dist-packages/numpy/core/fromnumeric.py:2499: VisibleDeprecationWarning: `rank` is deprecated; use the `ndim` attribute or function instead. To find the rank of a matrix see `numpy.linalg.matrix_rank`.
  VisibleDeprecationWarning)
Traceback (most recent call last):
  File "/usr/local/bin/pick_otus.py", line 999, in <module>
    main()
  File "/usr/local/bin/pick_otus.py", line 968, in main
    failure_path=failure_path)
  File "/usr/local/lib/python2.7/dist-packages/qiime/pick_otus.py", line 227, in __call__
    output_dir=dirname(abspath(result_path)))
  File "/usr/local/lib/python2.7/dist-packages/bfillings/sortmerna_v2.py", line 120, in build_database_sortmerna
    sdb = IndexDB(WorkingDir=output_dir, HALT_EXEC=HALT_EXEC)
  File "/usr/local/lib/python2.7/dist-packages/burrito/util.py", line 202, in __init__
    self._error_on_missing_application(params)
  File "/usr/local/lib/python2.7/dist-packages/burrito/util.py", line 467, in _error_on_missing_application
    "Is it in your path?" % command)
burrito.util.ApplicationNotFoundError: Cannot find indexdb_rna. Is it installed? Is it in your path?



======================================================================
ERROR: test_pick_subsampled_open_reference_otus_sortmerna_sumaclust_db (__main__.PickSubsampledReferenceOtusThroughOtuTableTests)
pick_subsampled_open_reference_otus functions as expected with
----------------------------------------------------------------------
Traceback (most recent call last):
  File "test_pick_open_reference_otus.py", line 911, in test_pick_subsampled_open_reference_otus_sortmerna_sumaclust_db
    output_dir=self.tmp_dir)
  File "/usr/local/lib/python2.7/dist-packages/bfillings/sortmerna_v2.py", line 120, in build_database_sortmerna
    sdb = IndexDB(WorkingDir=output_dir, HALT_EXEC=HALT_EXEC)
  File "/usr/local/lib/python2.7/dist-packages/burrito/util.py", line 202, in __init__
    self._error_on_missing_application(params)
  File "/usr/local/lib/python2.7/dist-packages/burrito/util.py", line 467, in _error_on_missing_application
    "Is it in your path?" % command)
ApplicationNotFoundError: Cannot find indexdb_rna. Is it installed? Is it in your path?

======================================================================
ERROR: test_pick_subsampled_open_reference_otus_sortmerna_sumaclust_step2_3 (__main__.PickSubsampledReferenceOtusThroughOtuTableTests)
pick_subsampled_open_reference_otus functions as expected with
----------------------------------------------------------------------
Traceback (most recent call last):
  File "test_pick_open_reference_otus.py", line 779, in test_pick_subsampled_open_reference_otus_sortmerna_sumaclust_step2_3
    minimum_failure_threshold=0)
  File "/usr/local/lib/python2.7/dist-packages/qiime/workflow/pick_open_reference_otus.py", line 715, in pick_subsampled_open_reference_otus
    close_logger_on_success=False)
  File "/usr/local/lib/python2.7/dist-packages/qiime/workflow/util.py", line 122, in call_commands_serially
    raise WorkflowError(msg)
WorkflowError:

*** ERROR RAISED DURING STEP: Pick Reference OTUs
Command run was:
 pick_otus.py -i /tmp/seqseg0qKc -o /tmp/qiime_wf_outRlEgRG/step1_otus -r /tmp/refseqsKjVf9T -m sortmerna  --suppress_new_clusters
Command returned exit status: 1
Stdout:

Stderr
/usr/local/lib/python2.7/dist-packages/numpy/core/fromnumeric.py:2499: VisibleDeprecationWarning: `rank` is deprecated; use the `ndim` attribute or function instead. To find the rank of a matrix see `numpy.linalg.matrix_rank`.
  VisibleDeprecationWarning)
Traceback (most recent call last):
  File "/usr/local/bin/pick_otus.py", line 999, in <module>
    main()
  File "/usr/local/bin/pick_otus.py", line 968, in main
    failure_path=failure_path)
  File "/usr/local/lib/python2.7/dist-packages/qiime/pick_otus.py", line 227, in __call__
    output_dir=dirname(abspath(result_path)))
  File "/usr/local/lib/python2.7/dist-packages/bfillings/sortmerna_v2.py", line 120, in build_database_sortmerna
    sdb = IndexDB(WorkingDir=output_dir, HALT_EXEC=HALT_EXEC)
  File "/usr/local/lib/python2.7/dist-packages/burrito/util.py", line 202, in __init__
    self._error_on_missing_application(params)
  File "/usr/local/lib/python2.7/dist-packages/burrito/util.py", line 467, in _error_on_missing_application
    "Is it in your path?" % command)
burrito.util.ApplicationNotFoundError: Cannot find indexdb_rna. Is it installed? Is it in your path?



======================================================================
ERROR: test_pick_subsampled_open_reference_otus_usearch (__main__.PickSubsampledReferenceOtusThroughOtuTableTests)
pick_subsampled_open_reference_otus functions as expected with
----------------------------------------------------------------------
Traceback (most recent call last):
  File "test_pick_open_reference_otus.py", line 497, in test_pick_subsampled_open_reference_otus_usearch
    minimum_failure_threshold=0)
  File "/usr/local/lib/python2.7/dist-packages/qiime/workflow/pick_open_reference_otus.py", line 675, in pick_subsampled_open_reference_otus
    close_logger_on_success=False)
  File "/usr/local/lib/python2.7/dist-packages/qiime/workflow/util.py", line 122, in call_commands_serially
    raise WorkflowError(msg)
WorkflowError:

*** ERROR RAISED DURING STEP: Pick Reference OTUs (prefilter)
Command run was:
 pick_otus.py -i /tmp/seqsLcwQ2Y -o /tmp/qiime_wf_outV7pyOE/prefilter_otus/ -r /tmp/refseqsSim3Tm -m usearch61_ref --similarity 0.6 --suppress_new_clusters
Command returned exit status: 1
Stdout:

Stderr
/usr/local/lib/python2.7/dist-packages/numpy/core/fromnumeric.py:2499: VisibleDeprecationWarning: `rank` is deprecated; use the `ndim` attribute or function instead. To find the rank of a matrix see `numpy.linalg.matrix_rank`.
  VisibleDeprecationWarning)
Traceback (most recent call last):
  File "/usr/local/bin/pick_otus.py", line 999, in <module>
    main()
  File "/usr/local/bin/pick_otus.py", line 892, in main
    otu_prefix=otu_prefix, HALT_EXEC=False)
  File "/usr/local/lib/python2.7/dist-packages/qiime/pick_otus.py", line 1800, in __call__
    HALT_EXEC=HALT_EXEC
  File "/usr/local/lib/python2.7/dist-packages/bfillings/usearch.py", line 1844, in usearch61_ref_cluster
    raise ApplicationError('Error running usearch61. Possible causes are '
burrito.util.ApplicationError: Error running usearch61. Possible causes are unsupported version (current supported version is usearch v6.1.544) is installed or improperly formatted input file was provided



----------------------------------------------------------------------
Ran 17 tests in 242.435s

FAILED (errors=5)
```

**Comment by wasade** [comment]
QIIME 1.8 tutorial does not work either due to removed URLs.

```bash

sh: 0: getcwd() failed: No such file or directory
--2015-01-11 11:10:54--  https://s3.amazonaws.com/s3-qiime_tutorial_files/moving_pictures_tutorial-1.8.0.tgz
Resolving s3.amazonaws.com (s3.amazonaws.com)... 54.231.9.232
Connecting to s3.amazonaws.com (s3.amazonaws.com)|54.231.9.232|:443... connected.
HTTP request sent, awaiting response... 404 Not Found
2015-01-11 11:10:54 ERROR 404: Not Found.

sh: 0: getcwd() failed: No such file or directory
--2015-01-11 11:10:54--  ftp://greengenes.microbio.me/greengenes_release/gg_13_5/gg_13_8_otus.tar.gz
           => ‘gg_13_8_otus.tar.gz’
Resolving greengenes.microbio.me (greengenes.microbio.me)... 128.138.93.17
Connecting to greengenes.microbio.me (greengenes.microbio.me)|128.138.93.17|:21... connected.
Logging in as anonymous ... Logged in!
==> SYST ... done.    ==> PWD ... done.
==> TYPE I ... done.  ==> CWD (1) /greengenes_release/gg_13_5 ... done.
==> SIZE gg_13_8_otus.tar.gz ... 320067060
==> PASV ... done.    ==> RETR gg_13_8_otus.tar.gz ... done.
gg_13_8_otus.tar.gz: No such file or directory
sh: 0: getcwd() failed: No such file or directory
tar (child): moving_pictures_tutorial-1.8.0.tgz: Cannot open: No such file or directory
tar (child): Error is not recoverable: exiting now
tar: Child returned status 2
tar: Error is not recoverable: exiting now
sh: 0: getcwd() failed: No such file or directory
tar (child): gg_13_8_otus.tar.gz: Cannot open: No such file or directory
tar (child): Error is not recoverable: exiting now
tar: Child returned status 2
tar: Error is not recoverable: exiting now
```

**Comment by wasade** [comment]
Downgrading is not easy...

**Comment by gregcaporaso** [comment]
@wasade, the pick_open_reference_otus.py test failures shouldn't be a problem (they're testing functionality that is not part of the base install, and which the tutorial doesn't hit by default). The best thing for us to do here would be for these tests to run conditionally, but that's not really something we could do for 1.9.0. 

Moving this discussion to IM for now...

**Comment by wasade** [comment]
Thanks, @gregcaporaso. Reason I put the `pick_open_reference_otus.py` tests in was to show that the relevant tests pass but the script is failing, so it isn't being caught by unit tests. I did not have a chance to run script level tests

**Comment by jairideout** [comment]
@wasade the numpy deprecation warnings are coming from PyCogent code (beta diversity I think), so there isn't an easy fix for this. It's strange that you're seeing those warnings with things like validate_mapping_file.py though. I'm not able to reproduce this locally with numpy 1.9.1.

Note that numpy >= 1.9.0 is required for filter_alignment.py, so you may encounter errors if using an older numpy version.

I created a separate issue for the numpy deprecation warnings so that we can investigate further prior to the release (#1850).

**Comment by jairideout** [comment]
@wasade I think this issue can be closed:

- deprecation warnings (#1850) will be fixed when #1854 is merged
- pick_open_reference_otus.py requiring h5py was fixed in #1853 (verified that both errors you reported are no longer raised)
- issues with the 1.9.0 moving pictures tarball should be resolved (@ElDeveloper updated the tarball hosted on ftp.microbio.me to the correct one)

Please reopen if there's anything I missed.
